api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -238,6 +238,52 @@ message HttpConnectionManager {
     google.protobuf.BoolValue enabled = 3;
   }
 
+  // Transformations that apply to path headers. The policy to apply these
+  // transformations can be tuned by the configuration field. Transformations
+  // can be applied internally only, so that the normalized header is used in
+  // matching, routing, and HTTP filters while the original path header is
```

Comment:
What about observability (logging, tracing)?

Reply:
It depends on the context. Only the normalized path is visible internally so if a filter logs headers, they will see the normalized header. If we finalize spans before forwarding to an upstream, we see the internal header, otherwise we see the forwarded header. This can obviously be changed in the code.

I'm not really sure what the right answer here is. If it should always be the "forwarded header" or always the original header, then we lose out on observability in some context. I think we either have to keep the logging context dependent as is, OR we allow `HttpTracerUtility` visibility into the original header (and for logs, I don't know -- as soon as we open the original header to logs, we open it up to filters).

e.g. if you log the request headers directly prior to forwarding, you will see the to-be-forwarded header, like here https://github.com/envoyproxy/envoy/blob/280654be74c4760a7f61f3658b035e4800f2deca/source/common/router/router.cc#L583. before that, you see the internal header.